### PR TITLE
Qualcomm AI Engine Direct - Quantizer refine for qat

### PR DESCRIPTION
**Pull request by chunit-quic**
Follow the instruction to resubmit the PR after [PR6513](https://github.com/pytorch/executorch/pull/6513) is reverted.

### CODE REVIEW

File: backends/qualcomm/quantizer/quantizer.py [hunk 1]
```diff
@@ -22,14 +22,17 @@
...
 
-from .utils import (
+from .annotators import OP_ANNOTATOR
+
+from .qconfig import (
+    get_16a16w_qnn_ptq_config,
     get_16a4w_qnn_ptq_config,
+    get_16a4w_qnn_qat_config,
     get_16a8w_qnn_ptq_config,
-    get_default_16bit_qnn_ptq_config,
```
Comment (cccclai) [inline backends/qualcomm/quantizer/quantizer.py:28]: Removing `get_default_16bit_qnn_ptq_config` causing internal failure...can we do it in a the following way
1. This PR includes both new config and the old config
2. I submit a PR internally to remove the old config call site
3. You submit a new PR to remove the old config

Then we can land it safely...

Comment (chunit-quic) [reply backends/qualcomm/quantizer/quantizer.py:28]: Hi Chen,
Thanks for pointing out what to fix.
It seems to fail again. Shuold I also add back `get_default_8bit_qnn_ptq_config`?

Comment (cccclai) [reply backends/qualcomm/quantizer/quantizer.py:28]: Sorry what is fail again?

Comment (chunit-quic) [reply backends/qualcomm/quantizer/quantizer.py:28]: Two  checks in CI chcek list below
image

Comment (cccclai) [reply backends/qualcomm/quantizer/quantizer.py:28]: Oh hmm I feel like the `Meta Internal-Only` check isn't very accurate...

Comment (chunit-quic) [reply backends/qualcomm/quantizer/quantizer.py:28]: No problem. Let me know what should be fix if any. :D

Comment (cccclai) [reply backends/qualcomm/quantizer/quantizer.py:28]: Let me import and check CI again...

**Comment by cccclai** [comment]
Oh looks like some internal failure is resolved! There is one more left but I think it's already there. I'll submit a PR to fix that and then merge this change.

**Review by tarun292 — APPROVED** [review]

**Comment by cccclai** [comment]
Hey can you rebase this change? I landed the internal PR...

File: backends/qualcomm/quantizer/observers/per_channel_param_observer.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import torch
+from torch.ao.quantization.observer import UniformQuantizationObserverBase
+
+
+# TODO move to torch/ao/quantization/observer.py.
+class PerChannelParamObserver(UniformQuantizationObserverBase):
```
Comment (kimishpatel) [inline backends/qualcomm/quantizer/observers/per_channel_param_observer.py:6]: Can you add some comments here as to why chose the min/max ranges in the way being done here?

Comment (chunit-quic) [reply backends/qualcomm/quantizer/observers/per_channel_param_observer.py:6]: Hi @kimishpatel, just add the comment. If it's not clear enought please feel free to let me know. Thanks

**Comment by cccclai** [comment]
Hey wonder if you can rebase this PR, I've been failing to import this PR and the error is following
```
    The Pull Request could not be imported cleanly from GitHub. This usually happens for one of two reasons:
    1. The Pull Request is out of date, if you rebase the PR on the latest commit on the GitHub branch and find merge conflicts this is probably the reason
    2. ShipIt is broken for this repo, check bunnylol `oss org/repo`, click on the ShipIt tab, and see if there are any alerts. Fix those and retry the import.
```

**Comment by chunit-quic** [comment]
> Hey wonder if you can rebase this PR, I've been failing to import this PR and the error is following

Just rebased. Thanks for pointing out.

**Comment by cccclai** [comment]
Thanks! A small `lintrunner` error...

**Pull request merged**